### DISCUSSION TITLE
A "Force Download" option on the in-game beatmap page

### DISCUSSION THREAD

**Discussion by Mornis457** [ppy/osu#31322] [Ideas]
There seems to be an issue with some maps appearing as unsubmitted. Those maps don't have an "Update" option. To fix those maps currently you have to copy the beatmap id or link, delete the mapset and then redowload it. Having a "force download" button in the in-game beatmap page that would automatically do that would help with that.

**Comment by Dekamir** [comment]
I want this for different reasons. Right now, I had a data corruption, which caused some of my beatmaps to disappear.
![osu_2025-10-29_14-42-59](https://github.com/user-attachments/assets/b6ca77d6-43ef-4c13-82c9-30b1d9c8b8ec)

Allowing to shift-click, or with any modifier (or hold on mobile) to force redownload the .osu file and reimport all difficulties to essentially repair the beatmap set would be awesome.
![osu_2025-10-29_14-43-05](https://github.com/user-attachments/assets/0cd1436a-f1d4-4e3f-b617-f85d4d0fc606)
